src/components/Table/TableCell.tsx

```diff
@@ -17,50 +11,69 @@ export enum TableCellType {
 
 export type TableCellProps = {
     cell: any;
-    state: TableState<any>;
+    selectionMode: string;
     type?: TableCellType;
+    isChecked?: boolean;
+    selectedRows: Key[];
+    setSelectedRows?: (ids?: Key[]) => void;
 };
 
-export const TableCell: FC<TableCellProps> = ({ cell, state, type = TableCellType.Default }) => {
+export const TableCell: FC<TableCellProps> = ({
+    cell,
+    selectionMode,
+    type = TableCellType.Default,
+    isChecked = false,
+    selectedRows,
+    setSelectedRows,
+}) => {
     const ref = useRef<HTMLTableCellElement | null>(null);
-    const { gridCellProps } = useTableCell({ node: cell }, state, ref);
-    const { checkboxProps } = useTableSelectionCheckbox({ key: cell.parentKey }, state);
-    const inputRef = useRef(null);
-    const {
-        inputProps: { checked },
-    } = useCheckbox(checkboxProps, useToggleState(checkboxProps), inputRef);
-    const { isFocusVisible, focusProps } = useFocusRing();
 
     if (type === TableCellType.Checkbox) {
         const { key } = cell;
+        const handleChange = () => {
+            if (!setSelectedRows) {
+                return;
+            }
+
+            if (isChecked) {
+                const filteredRows = selectedRows.filter((row) => row !== cell.parentKey);
+                setSelectedRows(filteredRows);
+                return;
+            }
+            if (selectionMode === SelectionMode.SingleSelect) {
+                setSelectedRows([cell.parentKey]);
+            } else {
+                setSelectedRows([...selectedRows, cell.parentKey]);
+            }
+        };
 
         return (
             <td
-                {...gridCellProps}
+                role="cell"
                 ref={ref}
                 className={merge([
                     'tw-pl-8 tw-py-4 tw-pr-4 tw-border-l-4',
-                    checked ? 'tw-border-violet-60' : 'tw-border-transparent',
+                    isChecked ? 'tw-border-violet-60' : 'tw-border-transparent',
                 ])}
                 data-test-id="table-select-cell"
             >
-                <Checkbox
+                <CheckboxCmp
```

Comment:
I'm curious, why was this renamed?

Reply:
I also don't like it :( But it's a condition in the quality checks to not have sonarLint warnings and there was a warning in the `Checkbox` enum value because something called `Checkbox` already was being imported.

Reply:
Renamed to `CheckboxComponent` 👍